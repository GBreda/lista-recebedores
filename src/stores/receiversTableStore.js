import { defineStore } from 'pinia'
import ReceiversService from '@/services/ReceiversService'

export const useReceiversTableStore = defineStore('receiversTableStore', {
  state: () => ({
    tableData: []
  }),
  actions: {
    async fetchTableData(payload) {
      try {
        const { data } = await ReceiversService.fetchReceivers(payload)

        this.tableData = data

        return Promise.resolve(data)
      } catch (error) {
        return Promise.reject(error)
      }
    }
  }
})
