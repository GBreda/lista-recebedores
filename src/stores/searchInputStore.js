import { defineStore } from 'pinia'

export const useSearchInputStore = defineStore('searchInputStore', {
  state: () => ({
    input: ''
  }),
  actions: {
    setSearchInputValue(value) {
      this.input = value
    }
  }
})
