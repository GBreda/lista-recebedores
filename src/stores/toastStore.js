import { defineStore } from 'pinia'

export const useToastStore = defineStore('toastStore', {
  state: () => ({
    toastInfo: {
      showToast: false,
      message: '',
      kind: ''
    }
  }),
  actions: {
    setToastInfo(value) {
      this.toastInfo = value
    }
  }
})
