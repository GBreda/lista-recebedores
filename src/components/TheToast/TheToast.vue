<template>
  <div class="the-toast" :class="kindClass">
    <p class="the-toast__message">{{ message }}</p>
    <img
      class="the-toast__close-icon"
      src="@/assets/images/blur-close-icon.png"
      alt="Fechar alerta"
      @click="closeToast"
    />
  </div>
</template>

<script setup>
import { computed } from 'vue'
import { useToastStore } from '@/stores/toastStore'

const toastStore = useToastStore()

const props = defineProps({
  kind: {
    type: String,
    validator(value) {
      return ['success', 'danger'].includes(value)
    },
    required: true
  },
  message: {
    type: String,
    required: true
  }
})

const kindClass = computed(() => `the-toast__${props.kind}`)

const closeToast = () => {
  toastStore.setToastInfo({
    showToast: false
  })
}
</script>

<style lang="scss" soped>
.the-toast {
  min-width: 28rem;
  display: flex;
  gap: 2rem;
  justify-content: space-between;
  align-items: center;
  padding: 2rem;
  border-radius: 0.8rem;
  position: absolute;
  top: 5.2rem;
  right: 2rem;

  &__message {
    color: $white;
    font-weight: 500;
  }

  &__close-icon {
    cursor: pointer;
  }

  &__success {
    background-color: $blue;
  }

  &__danger {
    background-color: $red;
  }
}
</style>
