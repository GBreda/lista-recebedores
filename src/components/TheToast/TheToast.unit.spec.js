import { describe, it, expect, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import TheToast from './TheToast.vue'
import { createTestingPinia } from '@pinia/testing'
import { useToastStore } from '@/stores/toastStore'

function mountTheToast({ props } = {}) {
  return mount(TheToast, {
    global: {
      plugins: [
        createTestingPinia({
          createSpy: vi.fn
        })
      ]
    },
    props: {
      ...props
    }
  })
}
describe('TheToast', () => {
  let wrapper

  it('should mount the component', () => {
    wrapper = wrapper = mountTheToast({
      props: {
        kind: 'success',
        message: 'Success Message'
      }
    })

    expect(wrapper.vm).toBeDefined()
  })

  it('should match the snapshot', () => {
    wrapper = wrapper = mountTheToast({
      props: {
        kind: 'success',
        message: 'Success Message'
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render with success style and correct message', () => {
    wrapper = mountTheToast({
      props: {
        kind: 'success',
        message: 'Success Message'
      }
    })

    expect(wrapper.find('.the-toast__success').exists()).toBe(true)
    expect(wrapper.find('.the-toast__message').text()).toBe('Success Message')
  })

  it('should render with danger style and correct message', () => {
    wrapper = mount(TheToast, {
      props: {
        kind: 'danger',
        message: 'Danger Message'
      }
    })

    expect(wrapper.find('.the-toast__danger').exists()).toBe(true)
    expect(wrapper.find('.the-toast__message').text()).toBe('Danger Message')
  })

  it('should call store action on toast close', async () => {
    wrapper = mount(TheToast, {
      props: {
        kind: 'success',
        message: 'Success Message'
      }
    })

    const store = useToastStore()

    await wrapper.find("[data-test-id='close-toast']").trigger('click')

    expect(store.setToastInfo).toHaveBeenCalled()
    expect(store.setToastInfo).toHaveBeenLastCalledWith({
      showToast: false
    })
  })
})
