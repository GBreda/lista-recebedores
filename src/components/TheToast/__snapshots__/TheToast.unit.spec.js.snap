// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TheToast > should match the snapshot 1`] = `
<div
  class="the-toast the-toast__success"
>
  <p
    class="the-toast__message"
  >
    Success Message
  </p>
  <img
    alt="Fechar alerta"
    class="the-toast__close-icon"
    data-test-id="close-toast"
    src="/src/assets/images/blur-close-icon.png"
  />
</div>
`;
